test/jdk/javax/accessibility/JSlider/JSliderAccessibilityTest.java

```diff
@@ -0,0 +1,248 @@
+            }
+        }
+
+        if (set.contains(AccessibleState.HORIZONTAL)) {
+            if (set.contains(AccessibleState.VERTICAL)) {
+                throw new RuntimeException("AccessibleStateSet shouldn't contain both HORIZONTAL and VERTICAL");
+            }
+            if (jSlider.getOrientation() != jSlider.HORIZONTAL) {
+                throw new RuntimeException("AccessibleStateSet contains HORIZONTAL but this Slider's orientation is not horizontal");
+            }
+        } else {
+            if (!set.contains(AccessibleState.VERTICAL)) {
+                throw new RuntimeException("AccessibleStateSet for a JSlider must contain HORIZONTAL or VERTICAL but this one does not");
+            }
+            if (jSlider.getOrientation() != jSlider.VERTICAL) {
+                throw new RuntimeException("AccessibleStateSet does not contain HORIZONTAL but this Slider's orientation is horizontal");
+            }
+        }
+
+        if (set.contains(AccessibleState.VERTICAL)) {
+            if (set.contains(AccessibleState.HORIZONTAL)) {
+                throw new RuntimeException("AccessibleStateSet shouldn't contain both HORIZONTAL and VERTICAL");
+            }
+            if (jSlider.getOrientation() != jSlider.VERTICAL) {
+                throw new RuntimeException("AccessibleStateSet contains VERTICAL but this Slider's orientation is not vertical");
+            }
+        } else {
+            if (!set.contains(AccessibleState.HORIZONTAL)) {
+                throw new RuntimeException("AccessibleStateSet for a JSlider must contain HORIZONTAL or VERTICAL but this one does not");
+            }
+            if (jSlider.getOrientation() != jSlider.HORIZONTAL) {
+                //System.err.println(this.set.toString());
+                throw new RuntimeException("AccessibleStateSet does not contain HORIZONTAL but this Slider's orientation is horizontal");
+            }
+        }
+
+        // Locale
+        Locale aLocale = accessibleContext.getLocale();
+        if (aLocale == null) {
+            throw new RuntimeException("AccessibleComponent.getLocale returned null");
+        }
+
+        Locale locale = jSlider.getLocale();
+        if (locale == null) {
+            throw new RuntimeException("JComponent.this.getLocale returned null");
+        }
+
+        if (!locale.equals(aLocale)) {
+            throw new RuntimeException("An accessible component should not have a different locale than the component it represents");
+        }
+
+        // AccessibleAction
+        AccessibleAction accessibleAction = accessibleContext.getAccessibleAction();
+        if (accessibleAction != null) {
+            if (accessibleAction.getAccessibleActionCount() != 2) {
+                throw new RuntimeException("Expected that getAccessibleActionCount value to be 2 but got " + accessibleAction.getAccessibleActionCount());
+            }
+
+            if (!accessibleAction.getAccessibleActionDescription(0).equals("increment")) {
```

Comment:
Instead of "string" the AccessibleAction.INCREMENT/DECREMENT constants should be used.